src/libsyntax/ext/base.rs

```diff
@@ -149,6 +149,33 @@ impl<F> MultiItemModifier for F
     }
 }
 
+pub trait ItemRenovator {
+    fn expand(&self,
+              ecx: &mut ExtCtxt,
+              span: Span,
+              meta_item: &ast::MetaItem,
+              item: Annotatable,
+              push: &mut FnMut(Annotatable))
+              -> Annotatable;
```

Comment:
Seems like we can drop the return type and just have implementors push all of the things they're creating/preserving into `push`?